File: rev/perplexed/reverse.py
def reverse_engineer_flag():
    # Encoded flag from the C code
    enc_flag = [
        0xE1, 0xA7, 0x1E, 0xF8, 0x75, 0x23, 0x7B, 0x61, 
        0xB9, 0x9D, 0xFC, 0x5A, 0x5B, 0xDF, 0x69, 0xD2, 
        0xFE, 0x1B, 0xED, 0xF4, 0xED, 0x67, 0xF4
    ]
    
    # Convert to binary representation
    enc_bits = []
    for byte in enc_flag:
        # Convert to 8-bit binary representation
        bits = [(byte >> i) & 1 for i in range(7, -1, -1)]
        enc_bits.extend(bits)
    
    # Reconstruct the flag
    flag_bits = []
    local_20 = 0
    
    for enc_bit in enc_bits:
        if local_20 == 0:
            local_20 = 1
        
        flag_bits.append(enc_bit)
        
        local_20 += 1
        if local_20 == 8:
            local_20 = 0
    
    # Convert bits to bytes
    flag_bytes = []
    for i in range(0, len(flag_bits), 8):
        byte = 0
        for j in range(8):
            if i + j < len(flag_bits):
                byte = (byte << 1) | flag_bits[i + j]
        flag_bytes.append(byte)
    
    # Convert bytes to ASCII
    flag = ''.join(chr(b) for b in flag_bytes)
    return flag

print(reverse_engineer_flag())

File: rev/perplexed/reverse.c
#include <stdio.h>
#include <string.h>

int main() {
    unsigned char enc_flag[] = {
        0xe1, 0xa7, 0x1e, 0xf8, 0x75, 0x23, 0x7b, 0x61,
        0xb9, 0x9d, 0xfc, 0x5a, 0x5b, 0xdf, 0x69, 0xd2,
        0xfe, 0x1b, 0xed, 0xf4, 0xed, 0x67, 0xf4
    };
    
    char flag[28] = {0}; // 27 chars + null terminator
    int flag_idx = 0;
    int bit_idx = 0;
    
    for (int i = 0; i < 23; i++) {
        for (int j = 0; j < 8; j++) {
            // Get the bit from enc_flag
            int bit = (enc_flag[i] >> (7 - j)) & 1;
            
            // Set the bit in our flag
            flag[flag_idx] |= (bit << (7 - bit_idx));
            
            bit_idx++;
            if (bit_idx == 8) {
                bit_idx = 0;
                flag_idx++;
            }
            
            if (flag_idx >= 27)
                break;
        }
    }
    
    printf("Flag: %s\n", flag);
    return 0;
}
